File: backend/src/main/java/com/acnh/critterpedia/service/TurnipService.java
package com.acnh.critterpedia.service;

import io.swagger.model.CalculationRequest;
import io.swagger.model.TurnipPatternMap;

public interface TurnipService {
    void pregeneratePatterns();

    TurnipPatternMap getPossiblePatterns(CalculationRequest calculationRequest);
}

File: frontend/src/app/shared/api/turnip.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { Observable } from 'rxjs';
import { TurnipCalculationRequest } from '../model/turnipCalculationRequest';
import { TurnipPatternMap } from '../model/turnipPatternMap';

@Injectable({
  providedIn: 'root'
})
export class TurnipService {
  private basePath = `${environment.basePath}/turnipPattern/`;

  constructor(private httpClient: HttpClient) { }

  getPatterns(calculationRequest: TurnipCalculationRequest): Observable<TurnipPatternMap> {
    return this.httpClient.post<TurnipPatternMap>(this.basePath, calculationRequest);    
  }
}
